core/commands/partials/vagrant_configuration_generator.rb

```diff
@@ -298,10 +294,27 @@ def get_role_description(name, product, box)
                        {}
                      end
     @ui.info("Recipe #{recipe_name}")
-    make_role_json(name, product_config, recipe_name, box)
+    { recipe: recipe_name, config: product_config }
   end
   # rubocop:enable Metrics/MethodLength
 
+  # Generate the role description for the specified node.
+  #
+  # @param name [String] internal name of the machine specified in the template
+  # @param products [Array<Hash>] list of parameters of products to configure from configuration file
+  # @param box [String] name of the box
+  # @return [String] pretty formatted role description in JSON format
+  def get_role_description(name, products, box)
+    products_configs = {}
+    recipes_names = []
+    products.each do |product|
+      recipe_and_config = make_product_config_and_recipe_name(product, box)
+      products_configs.merge!(recipe_and_config[:config])
```

Comment:
I presume that we can not install 2 recipes that require configuration file? I.e. in our recipes use the same name for the path to the configuration file. When merging, only the last one will remain.